be/src/util/frame_of_reference_coding.cpp

```diff
@@ -72,6 +72,168 @@ void ForEncoder<T>::put_batch(const T* in_data, size_t count) {
 
 // todo(kks): improve this method by SIMD instructions
 
+template <typename T>
+void ForEncoder<T>::bit_pack_8(const T* input, uint8_t in_num, int bit_width, uint8_t* output) {
```

Comment:
in`bit_pack_...` functions we have many same codes. can you combine these functions using template? some differentiated logic can be determined according to template parameters.

Reply:
well it may be not feasible. you can decide.